certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/validation/model/ElementValidationDateRangeList.java

```diff
@@ -0,0 +1,210 @@
+  }
+
+  private List<ValidationError> getChildValidationErrors(ElementData data,
+      Optional<ElementId> categoryId, ElementValueDateRangeList dateRangeList) {
+    if (dateRangeList.dateRangeList() == null) {
+      return Collections.emptyList();
+    }
+
+    final var incompleteErrors = getIncompleteDateRangeErrors(data, categoryId, dateRangeList);
+    final var incorrectErrors = getIncorrectDateRangeErrors(data, categoryId, dateRangeList);
+
+    return Stream
+        .concat(incompleteErrors.stream(), incorrectErrors.stream())
+        .toList();
+  }
+
+  private List<ValidationError> getOverlappingErrors(ElementData data,
+      Optional<ElementId> categoryId, ElementValueDateRangeList dateRangeList) {
+    if (dateRangeList.dateRangeList() == null) {
+      return Collections.emptyList();
+    }
+
+    return dateRangeList.dateRangeList().stream()
+        .filter(dateRange -> doesDateRangeHaveOverlap(dateRange, dateRangeList.dateRangeList()))
+        .findFirst()
+        .map(dateRange -> errorMessage(
+                data,
+                dateRangeList.dateRangeListId(),
+                categoryId,
+                "Ange sjukskrivningsperioder som inte överlappar varandra."
+            )
+        )
+        .stream().toList();
+
+
+  }
+
+  private boolean doesDateRangeHaveOverlap(DateRange dateRange, List<DateRange> list) {
+    return list.stream()
+        .filter(comparedDateRange -> comparedDateRange.dateRangeId() != dateRange.dateRangeId())
+        .anyMatch(comparedDateRange -> doesDateRangesOverlap(comparedDateRange, dateRange));
+  }
+
+  private boolean doesDateRangesOverlap(DateRange dateRange1, DateRange dateRange2) {
+    final var overlap = Math.min(dateRange1.to().toEpochDay(), dateRange2.to().toEpochDay()) -
+        Math.max(dateRange1.from().toEpochDay(), dateRange2.from().toEpochDay());
+
+    return overlap >= 0;
+  }
+
+  private List<ValidationError> getIncorrectDateRangeErrors(ElementData data,
+      Optional<ElementId> categoryId,
+      ElementValueDateRangeList dateRangeList) {
+    return dateRangeList.dateRangeList().stream()
+        .filter(this::isDateRangeIncorrect)
+        .map(dateRange -> errorMessage(
+                data, getFieldId(dateRange.dateRangeId(), RANGE_SUFFIX), categoryId,
+                "Ange ett slutdatum som infaller efter startdatumet."
+            )
+        )
+        .toList();
+  }
+
+  private List<ValidationError> getIncompleteDateRangeErrors(ElementData data,
+      Optional<ElementId> categoryId, ElementValueDateRangeList dateRangeList) {
+    return dateRangeList.dateRangeList().stream()
+        .filter(dateRange -> !isDateRangeComplete(dateRange))
+        .map(dateRange -> errorMessage(
+                data, getFieldIdOfIncompleteDateRange(dateRange), categoryId, "Ange ett datum."
+            )
+        )
+        .toList();
+  }
+
+  private ElementValueDateRangeList getValue(ElementValue value) {
+    if (value == null) {
```

Comment:
Redundant check since its already done earlier in the code

Reply:
Data is checked if it's null before but not value. This is the pattern we followed with the other validation. Which line is it that you mean it's checked before?

Reply:
My bad, keep this as it is! :)